model_analyzer/analyzer.py

```diff
@@ -223,6 +241,42 @@ def _profile_models(self):
                 finally:
                     self._state_manager.save_checkpoint()
 
+    def _analyze_models(self):
+        # TODO: Until we get rid of analysis we need to copy some values from profile
+        self._config._fields["analysis_models"] = self._config._fields[
+            "profile_models"]
+
+        gpu_info = self._state_manager.get_state_variable('MetricsManager.gpus')
+        if not gpu_info:
+            gpu_info = {}
+
+        self._result_manager.compile_and_sort_results()
+
+    def _create_summary_tables(self, verbose: bool):
+        self._result_table_manager = ResultTableManager(self._config,
+                                                        self._result_manager)
+
+        self._result_table_manager.create_tables()
+        self._result_table_manager.tabulate_results()
+        self._result_table_manager.export_results()
+
+        if verbose:
+            self._result_table_manager.write_results()
+
+    def _create_summary_reports(self, mode: str):
+        gpu_info = self._state_manager.get_state_variable('MetricsManager.gpus')
```

Comment:
Let's avoid reaching into state manager though a hardcoded string. Ideally this goes through a public function in ResultManager, although this path seems to be MetricsManager?

Reply:
I've filed TMA-809 to track this work